### PR TITLE
Use couch_server:delete_file on view cleanup

### PR DESCRIPTION
**Pull request by iilyak**
This PR depends on:
- https://github.com/apache/couchdb-couch/pull/115

### CODE REVIEW

**Comment by eiri** [comment]
Why in-module tests? MRView has separate test suites for the rest of the modules.

File: src/couch_mrview_cleanup.erl [hunk 1]
```diff
@@ -41,7 +41,52 @@ run(Db) ->
...
+    try
+        {RootDir, ViewFile} = setup(false),
+        ?assert(filelib:is_regular(ViewFile)),
+        delete_view_file(RootDir, ViewFile),
+        ?assert(not filelib:is_regular(ViewFile)),
+        RenamedViewFile = couch_server:renamed_filename(ViewFile),
+        ?assert(not filelib:is_regular(RenamedViewFile)),
+        ok
+    after
+        meck:unload(config)
```
Comment (rnewson) [inline src/couch_mrview_cleanup.erl]: can we use setup/teardown in the usual eunit way?

File: src/couch_mrview_cleanup.erl [hunk 2]
```diff
@@ -41,7 +41,52 @@ run(Db) ->
...
+    ok = couch_file:init_delete_dir(RootDir),
+    ok = file:write_file(ViewFile, <<>>),
+    {RootDir, ViewFile}.
+
+remove_on_delete_test() ->
+    try
+        {RootDir, ViewFile} = setup(true),
+        ?assert(filelib:is_regular(ViewFile)),
+        delete_view_file(RootDir, ViewFile),
+        ?assert(not filelib:is_regular(ViewFile)),
```
Comment (rnewson) [inline src/couch_mrview_cleanup.erl]: use `?assertNot(`

**Comment by iilyak** [comment]
@eiri: > Why in-module tests? MRView has separate test suites for the rest of the modules.
It is in the module tests because I'm testing private function.

**Comment by eiri** [comment]
@iilyak: ah, fair enough. I suppose we could've use something like `{eunit_compile_opts, [export_all]}` in rebar config to not to mix things, but this is out of the scope.

**Comment by rnewson** [comment]
+1

File: src/couch_mrview_cleanup.erl [hunk 3]
```diff
@@ -41,7 +41,71 @@ run(Db) ->
...
+    }.
+
+should_rename_on_delete(_, {RootDir, ViewFile}) ->
+    ?_test(begin
+        ?assert(filelib:is_regular(ViewFile)),
+        Result = delete_view_file(RootDir, ViewFile),
+        ?assertMatch({ok, {renamed, _}}, Result),
+        {ok, {renamed, RenamedViewFile}} = Result,
+        ?assertNot(filelib:is_regular(ViewFile)),
+        ?assert(filelib:is_regular(RenamedViewFile))
```
Comment (rnewson) [inline src/couch_mrview_cleanup.erl]: much clearer.

**Comment by rnewson** [comment]
+1

**Comment by iilyak** [comment]
squashed

**Pull request merged**